test/react-native/features/native-stack.feature

```diff
@@ -1,66 +1,66 @@
-#   And the event "exceptions.0.type" equals "reactnativejs"
-#   And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
+@android_only
+Scenario: Handled JS error with native stacktrace
+  When I run "NativeStackHandledScenario"
+  Then I wait to receive a request
+  And the event "unhandled" is false
+  And the payload field "events.0.exceptions" is an array with 1 elements
+  And the event "exceptions.0.errorClass" equals "Error"
+  And the event "exceptions.0.message" equals "NativeStackHandledScenario"
+  And the event "exceptions.0.type" equals "reactnativejs"
+  And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
 
-#   # validate the native error
-#   And the event "exceptions.1.errorClass" equals "Error"
-#   And the event "exceptions.1.message" equals "NativeStackHandledScenario"
-#   And the event "exceptions.1.type" equals "android"
-#   And the payload field "events.0.exceptions.1.stacktrace" is a non-empty array
-#   And the payload field "events.0.exceptions.1.stacktrace.0.file" equals "Scenario.kt"
-#   And the payload field "events.0.exceptions.1.stacktrace.1.file" equals "NativeStackHandledScenario.kt"
-#   And the payload field "events.0.exceptions.1.stacktrace.2.file" equals "BugsnagModule.java"
-#   And the payload field "events.0.exceptions.1.stacktrace.0.method" equals one of:
-#     | com.reactnative.scenarios.Scenario.generateException |
-#     | generateException |
-#   And the payload field "events.0.exceptions.1.stacktrace.1.method" equals one of:
-#     | com.reactnative.scenarios.NativeStackHandledScenario.run |
-#     | run |
-#   And the payload field "events.0.exceptions.1.stacktrace.2.method" equals one of:
-#     | com.reactnative.module.BugsnagModule.runScenario |
-#     | runScenario |
+  # the native part of the stack comes first
+  And the payload field "events.0.exceptions.0.stacktrace.0.type" equals "android"
+  And the payload field "events.0.exceptions.0.stacktrace.0.file" equals "Scenario.kt"
+  And the payload field "events.0.exceptions.0.stacktrace.0.method" equals one of:
+    | com.reactnative.scenarios.Scenario.generateException |
+    | generateException |
+  And the payload field "events.0.exceptions.0.stacktrace.1.method" equals one of:
+    | com.reactnative.scenarios.NativeStackHandledScenario.run |
+    | run |
+  And the payload field "events.0.exceptions.0.stacktrace.2.method" equals one of:
+    | com.reactnative.module.BugsnagModule.runScenario |
+    | runScenario |
 
-#   # PLAT-5117 addresses float serialization
-#   And the payload field "events.0.exceptions.1.stacktrace.0.lineNumber" equals 1
-#   And the payload field "events.0.exceptions.1.stacktrace.1.lineNumber" equals 1
-#   And the payload field "events.0.exceptions.1.stacktrace.2.lineNumber" equals 2
+  # the javascript part follows
+  And the payload field "events.0.exceptions.0.stacktrace.10.columnNumber" is not null
```

Comment:
This seems to fail on RN 0.63 as there are 13 native stackframes. Changing the frame index to 14 might work in the short term, or it might be possible to create some step that looks at the last frame in a stacktrace?

Reply:
Ahh that's frustrating! The last two stackframes don't have much info so I think I'll bump this to the 14th item as you suggest.